service/worker/batcher/activities.go

```diff
@@ -117,7 +117,7 @@ func (a *activities) BatchActivity(ctx context.Context, batchParams BatchParams)
 		executions := batchParams.Executions
 		pageToken := hbd.PageToken
 		if len(batchParams.Query) > 0 {
-			resp, err := sdkClient.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			resp, err := sdkClient.ScanWorkflow(ctx, &workflowservice.ScanWorkflowExecutionsRequest{
```

Comment:
@cretz Could you help me understand why we have ListWorkflow and ScanWorkflow here? Also, why don't we just follow the AIP standard of `ListWorkflowExecutions`?

Reply:
Historical compatibility reasons. We don't make this mistake in newer APIs. We don't even offer Scan and I suspect many on server team would suggest we do not use it. Can you confirm we this PR is invalid and we should be using list not scan?

Reply:
> Also, AIPs don't apply to Go API. Did you mean API? We name things in proto differently than SDKs for many reasons.

Oh never mind. I assumed this was a wrapper for an API of the same name. This looks fine to me.

> I am a bit confused on scan vs list use in this PR. Are we saying list is good enough for users but not good enough for us? If list is too slow, can we address that instead of not using it? Should we expose "scan" in other SDKs because we expect SDK users to use it when list is too slow? How would we tell a user to write their own batch/multi call using the SDK?

@samanbarghi care to comment here?

Reply:
The issue with List API currently is that it uses some sort of `ORDER BY` either provided by the client or enforced on the server side. This proved to be very slow for deep pagination of large number of documents in Elastic Search, it gets exponentially slower as we fetch consequent pages. 
Scan API uses the default ordering of the documents in ES which provides constant latency for paginating through large number of documents. So for queries that iterate through all documents but do not need specific order (e.g., migration) it's much more efficient to use Scan API, but for any query that requires some sort of ordering List API should be used. 

Also, @rodrigozhou mentioned that he had some new findings and it might not be necessary to replace List with Scan for functions in this PR. So I put this PR on hold for now until we are sure we want to move forward.

Reply:
:+1: In SDKs we have been under the impression that list is good enough for users. If not, we may need to make changes to our approach.

Reply:
@samanbarghi If it's on hold, could you put it in draft mode and then republish it when ready? That makes things easier on the review inbox side

Reply:
Done